### PR TITLE
enhance informer

### PR DESCRIPTION
**Pull request by rjbhewei**
1. informer list request support pagination like client-go(default limit 500).  

    eg: PageInformerExample.java

    fix issue : https://github.com/kubernetes-client/java/issues/628

2. informer local-cache filter resource fields which customer definition，to ease server-side JVM memory usage，avoid oom.

    eg: FieldFilterPageInformerExample.java

### CODE REVIEW

File: extended/pom.xml [hunk 1]
```diff
@@ -136,10 +152,14 @@
         </plugins>
     </build>
     <properties>
+        <maven.compiler.target>1.7</maven.compiler.target>
```
Comment (brendandburns) [inline extended/pom.xml]: Please preserve the existing parts w/ the variables as much as possible.

File: extended/src/main/java/io/kubernetes/client/extended/informer/FilterField.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package io.kubernetes.client.extended.informer;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class FilterField {
+
+  private String name;
+
```
Comment (brendandburns) [inline extended/src/main/java/io/kubernetes/client/extended/informer/FilterField.java]: Spacing is weird here. Remove extra newlines?

File: extended/src/main/java/io/kubernetes/client/extended/informer/PageSharedInformerFactory.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  public PageSharedInformerFactory(ApiClient client, ExecutorService threadPool) {
+    super(client, threadPool);
+  }
+
+  protected <ApiType, ApiListType> ListerWatcher<ApiType, ApiListType> listerWatcherFor(
+      final CallGenerator callGenerator,
+      final Class<ApiType> apiTypeClass,
+      final Class<ApiListType> apiListTypeClass) {
+
+    if (apiClient.getHttpClient().readTimeoutMillis() > 0) {
```
Comment (brendandburns) [inline extended/src/main/java/io/kubernetes/client/extended/informer/PageSharedInformerFactory.java]: Hrm, do we want to do this? Or do we want throw with `IlleagalArgumentException` silently changing the timeout feels weird to me.

File: extended/src/main/java/io/kubernetes/client/extended/informer/PageSharedInformerFactory.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      // set read timeout zero to ensure client doesn't time out
+      OkHttpClient httpClient =
+          apiClient.getHttpClient().newBuilder().readTimeout(0, TimeUnit.MILLISECONDS).build();
+      apiClient.setHttpClient(httpClient);
+    }
+
+    return new ListerWatcher<ApiType, ApiListType>() {
```
Comment (brendandburns) [inline extended/src/main/java/io/kubernetes/client/extended/informer/PageSharedInformerFactory.java]: This feels like a pretty big chunk of code to be an anonymous class, perhaps break it out as it's own class? It's pretty hard to test it as is.

**Comment by brendandburns** [comment]
The Field Filter code seems separate from the paging code. Can you split this into 2 PRs?

Thanks

**Comment by fejta** [comment]
/hold
testing

**Comment by fejta** [comment]
/hold cancel
thanks

File: extended/src/main/java/io/kubernetes/client/extended/informer/PageSharedInformerFactory.java (hunk 4 above)
Comment (rjbhewei) [reply extended/src/main/java/io/kubernetes/client/extended/informer/PageSharedInformerFactory.java]: same with [SharedInformerFactory](https://github.com/kubernetes-client/java/blob/69ca6fd7cca4f8188ec9b490530e53cc9b0dea09/util/src/main/java/io/kubernetes/client/informer/SharedInformerFactory.java#L131-L136)

File: extended/src/main/java/io/kubernetes/client/extended/informer/PageSharedInformerFactory.java (hunk 3 above)
Comment (rjbhewei) [reply extended/src/main/java/io/kubernetes/client/extended/informer/PageSharedInformerFactory.java]: same with [SharedInformerFactory](https://github.com/kubernetes-client/java/blob/69ca6fd7cca4f8188ec9b490530e53cc9b0dea09/util/src/main/java/io/kubernetes/client/informer/SharedInformerFactory.java#L131-L136)

File: extended/src/main/java/io/kubernetes/client/extended/informer/PageSharedInformerFactory.java (hunk 4 above)
Comment (rjbhewei) [reply extended/src/main/java/io/kubernetes/client/extended/informer/PageSharedInformerFactory.java]: PageSharedInformerFactory  and PagerCallGeneratorParams use like client-go, add page default limit size (500)，this factory use for scale data。

I wanted to enhance the new features by changing the original code as little as possible.Without affecting the original function。

submit the basic implementation first, and then change other related implementations。

**Comment by yue9944882** [comment]
plz split the pull by their intention

**Pull request closed without merging**